File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteJobAsset/QuoteJobAsset.cs
using System.Collections;

namespace FSM.Entity
{
    namespace QuoteJobAssets
    {
        public class QuoteJobAsset
        {
            private DataTypeValidator _dataTypeValidator;

            public QuoteJobAsset()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _QuoteJobAssetID = 0;

            public int QuoteJobAssetID
            {
                get
                {
                    return _QuoteJobAssetID;
                }
            }

            public object SetQuoteJobAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteJobAssetID", value);
                }
            }

            private int _QuoteJobID = 0;

            public int QuoteJobID
            {
                get
                {
                    return _QuoteJobID;
                }
            }

            public object SetQuoteJobID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteJobID", value);
                }
            }

            private int _AssetID = 0;

            public int AssetID
            {
                get
                {
                    return _AssetID;
                }
            }

            public object SetAssetID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AssetID", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteJobAsset/QuoteJobAsset.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteJobAssets

        Public Class QuoteJobAsset

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

#End Region

#Region "QuoteJobAsset Properties"


            Private _QuoteJobAssetID As Integer = 0
            Public ReadOnly Property QuoteJobAssetID() As Integer
                Get
                    Return _QuoteJobAssetID
                End Get
            End Property
            Public WriteOnly Property SetQuoteJobAssetID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteJobAssetID", Value)
                End Set
            End Property


            Private _QuoteJobID As Integer = 0
            Public ReadOnly Property QuoteJobID() As Integer
                Get
                    Return _QuoteJobID
                End Get
            End Property
            Public WriteOnly Property SetQuoteJobID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteJobID", Value)
                End Set
            End Property


            Private _AssetID As Integer = 0
            Public ReadOnly Property AssetID() As Integer
                Get
                    Return _AssetID
                End Get
            End Property
            Public WriteOnly Property SetAssetID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AssetID", Value)
                End Set
            End Property



#End Region

        End Class

    End Namespace

End Namespace

